hooks/tk-flame_actions.py

```diff
@@ -249,125 +281,446 @@ def _import_batch_group_from_shot(self, sg_publish_data):
+        Query the action_mappings entry to get every Published Type that's considered as a clip
 
-        :param path: The path containing the formatting character.
-        :param frame_range: Frame range, two ints in a str separated by a '-'
-        :type path: str
-        :type frame_range: str
+        :return: List of Published Type
+        :rtype: [str]
+        """
+
+        return [entry[0] for entry in self.parent.get_setting("action_mappings", {}).items() if
+                CLIP_ACTION in entry[1]]
 
-        :return: The path with the frame range.
+    @property
+    def supported_batch_types(self):
+        """
+        Query the action_mappings entry to get every Published Type that's considered as a Batch file
+
+        :return: List of Published Type
+        :rtype: [str]
+        """
+
+        return [entry[0] for entry in self.parent.get_setting("action_mappings", {}).items() if
+                BATCH_ACTION in entry[1]]
+
+    @property
+    def import_location(self):
+        """
+        Schematic Reel where the loader should import the clips.
+
+        :return: Location to import clip
         :rtype: str
         """
-        # Gets the two range numbers we need
-        ranges = [int(x) for x in str.split(frame_range, '-')]
 
-        # Checks that we have the info we need
-        if not ranges or len(ranges) != 2:
-            raise Exception("No ranges found")
+        return "Schematic Reel 1"
 
-        # Cuts off everything after the position of the formatting char.
-        path_end = path[path.find('%'):]
+    @property
+    def want_write_file_node(self):
+        """
+        Define if we want to link a Write File node to a new Batch group
 
-        # Get the formatting alone
-        formatting_str = path_end[:path_end.find('d') + 1]
+        Flame 2018.3 or above is needed for this functionality
 
-        # Gets the formatted frames numbers
-        start_frame = formatting_str % int(ranges[0])
-        end_frame = formatting_str % int(ranges[1])
+        :return: Hint about using a linking a Write File node
+        :rtype: bool
+        """
 
-        # Generates back the frame range, now formatted
-        frame_range = "[{}-{}]".format(
-            start_frame, end_frame
-        )
+        if sgtk.platform.current_engine().is_version_less_than("2018.3"):
+            return False
+
+        return True
+
+    @property
+    def use_template(self):
+        """
+        Define if we want to use templates to generate de Write File node attribute.
+
+        :return: Hint about using the templates to generate the Write File node attribute.
+        :rtype: bool
+        """
+
+        return True
+
+    @property
+    def media_path_root(self):
+        """
+        This path helps to setup the write_file_node by specifying where to write the medias.
+
+        :return: Media path root
+        :type: str
+        """
+        return "/var/tmp"
```

Comment:
This works, but some of these properties could also be done as a class attributes. Would make it easier for subclasses to override.

```python
class FlameActions(HookBaseClass):

    media_path_root = "/var/tmp"
    media_path_pattern = "<shot name>_{segment_name}_v<version>.<frame>"
    media_file_type = "OpenEXR"
    ...
```

Just a thought.